app/controllers/ems_common.rb

```diff
@@ -165,6 +165,39 @@ def edit
                     :url  => "/#{controller_name}/#{@ems.id}/edit")
   end
 
+  def properties
+    # Get the query parameters:
+    id = params[:id]
+    type = params[:type]
+
+    # Assert the permissions:
+    permission_suffix = if id.blank? then :new else :edit end
+    assert_privileges("#{permission_prefix}_#{permission_suffix}")
+
+    # Check the id:
+    if id.blank?
```

Comment:
isnt this a common practice (e.g. repeats itself?) maybe make sense to extract to a common method.

Reply:
I am not sure how common this is. Looks like the subject of a different pull request to me. I will not include this in the next version of the pull request. Let me know if you consider it a blocker for this pull request.